### ISSUE TITLE
feat(barcode-scanning): support all barcode types

### ISSUE THREAD

**Issue by robingenz** [capawesome-team/capacitor-mlkit#105]
### Plugin(s)

- [X] Barcode Scanning
- [ ] Face Detection
- [ ] Face Mesh Detection
- [ ] Selfie Segmentation
- [ ] Translation

### Current problem

Currently, only the `displayValue`/`rawValue` of a barcode is returned.
However, the ML Kit Barcode Scanning SDK also offers the option of returning parsed data such as [calendar event details](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode#public-barcode.calendarevent-getcalendarevent) or [email details](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode#public-barcode.email-getemail).

### Preferred solution

Extend the barcode interface with the following properties:

- [CalendarEvent](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.CalendarEvent)
- [ContactInfo](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.ContactInfo)
- [DriverLicense](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.DriverLicense)
- [Email](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.Email)
- [GeoPoint](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.GeoPoint)
- [Phone](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.Phone)
- [Sms](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.Sms)
- [UrlBookmark](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.UrlBookmark)
- [WiFi](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode.WiFi)

Example:

```ts
export interface Barcode {
  ...
  wifi?: BarcodeWifi;
}

export interface BarcodeWifi {
  encryptionType: BarcodeWifiEncryptionType;
  password: string;
  ssid: string;
}

export enum BarcodeWifiEncryptionType {
  OPEN = 1,
  WPA = 2,
  WEP = 3
}
```

### Before submitting

- [X] I have read and followed the [feature request guidelines](https://capawesome.io/contributing/feature-requests/).
- [X] I have attached links to possibly related issues and discussions.

**Comment by mertyldrr** [comment]
@robingenz 👋  could you pls assign this issue to me?

**Comment by robingenz** [comment]
Closed by #206
